modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GMLEncodingUtils.java

```diff
@@ -260,31 +276,39 @@ public XSDTypeDefinition createXmlTypeFromFeatureType(SimpleFeatureType featureT
             if ( attribute instanceof GeometryDescriptor ) {
                 Class binding = attribute.getType().getBinding();
                 if ( Point.class.isAssignableFrom( binding ) ) {
```

Comment:
Is it possible to use Java 7 support for String switch statements here combined with Geometry.getGeometryType() or similar?